src/backend/distributed/utils/maintenanced.c

```diff
@@ -194,11 +199,11 @@ InitializeMaintenanceDaemonBackend(void)
 		}
 
 		dbData->daemonStarted = true;
-		dbData->workerPid = 0;
 		dbData->triggerMetadataSync = false;
-		LWLockRelease(&MaintenanceDaemonControl->lock);
+		WaitForBackgroundWorkerStartup(handle, &dbData->workerPid);
```

Comment:
My fear is after `LWLockRelease` & before the worker sets `workerPid` someone drops db or something & that causes a leftover worker. Creation/destruction of maintenance daemon should be rare enough that we don't need to be lock ricers